hexapod_controller/include/hexapod_controller/hexapod_controller_simple.hpp

```diff
@@ -18,18 +18,181 @@ namespace hexapod_controller {
         typedef std::array<double, ARRAY_DIM> array_t;
 
         HexapodControllerSimple() {}
-        HexapodControllerSimple(const std::vector<double>& ctrl, std::vector<int> broken_legs);
+        HexapodControllerSimple(const std::vector<double>& ctrl, std::vector<int> broken_legs) : _broken_legs(broken_legs)
+        {
+            set_parameters(ctrl);
+        }
 
-        void set_parameters(const std::vector<double>& ctrl);
-        std::vector<double> parameters();
+        void set_parameters(const std::vector<double>& ctrl)
+        {
+            assert(ctrl.size() == 36);
 
-        void set_broken(const std::vector<int> broken_legs);
-        std::vector<int> broken_legs();
+            _legs0commands.clear();
+            _legs1commands.clear();
+            _legs2commands.clear();
+            _legs3commands.clear();
+            _legs4commands.clear();
+            _legs5commands.clear();
 
-        std::vector<double> pos(double t);
+            _controller = ctrl;
+
+            _legs0commands.push_back(_control_signal(ctrl[0], ctrl[1], ctrl[2]));
+            _legs0commands.push_back(_control_signal(ctrl[3], ctrl[4], ctrl[5]));
+            _legs0commands.push_back(_control_signal(ctrl[3], ctrl[4], ctrl[5]));
+
+            _legs1commands.push_back(_control_signal(ctrl[6], ctrl[7], ctrl[8]));
+            _legs1commands.push_back(_control_signal(ctrl[9], ctrl[10], ctrl[11]));
+            _legs1commands.push_back(_control_signal(ctrl[9], ctrl[10], ctrl[11]));
+
+            _legs2commands.push_back(_control_signal(ctrl[12], ctrl[13], ctrl[14]));
+            _legs2commands.push_back(_control_signal(ctrl[15], ctrl[16], ctrl[17]));
+            _legs2commands.push_back(_control_signal(ctrl[15], ctrl[16], ctrl[17]));
+
+            _legs3commands.push_back(_control_signal(ctrl[18], ctrl[19], ctrl[20]));
+            _legs3commands.push_back(_control_signal(ctrl[21], ctrl[22], ctrl[23]));
+            _legs3commands.push_back(_control_signal(ctrl[21], ctrl[22], ctrl[23]));
+
+            _legs4commands.push_back(_control_signal(ctrl[24], ctrl[25], ctrl[26]));
+            _legs4commands.push_back(_control_signal(ctrl[27], ctrl[28], ctrl[29]));
+            _legs4commands.push_back(_control_signal(ctrl[27], ctrl[28], ctrl[29]));
+
+            _legs5commands.push_back(_control_signal(ctrl[30], ctrl[31], ctrl[32]));
+            _legs5commands.push_back(_control_signal(ctrl[33], ctrl[34], ctrl[35]));
+            _legs5commands.push_back(_control_signal(ctrl[33], ctrl[34], ctrl[35]));
+        }
+
+        std::vector<double> parameters()
```

Comment:
According to good practice, getters should return const references and should be declared const when possible. We might replace it with

```C++
const std::vector<double> parameters() const
```

Same goes for `broken_legs()`.